models/propertyvalue.go

```diff
@@ -19,37 +19,39 @@ import (
-	Shift        string `json:"shift" yaml:"shift,omitempty"`               // Shift to be applied after masking, prior to get/set of property
-	Scale        string `json:"scale" yaml:"scale,omitempty"`               // Multiplicative factor to be applied after shifting, prior to get/set of property
-	Offset       string `json:"offset" yaml:"offset,omitempty"`             // Additive factor to be applied after multiplying, prior to get/set of property
-	Base         string `json:"base" yaml:"base,omitempty"`                 // Base for property to be applied to, leave 0 for no power operation (i.e. base ^ property: 2 ^ 10)
-	Assertion    string `json:"assertion" yaml:"assertion,omitempty"`       // Required value of the property, set for checking error state.  Failing an assertion condition will mark the device with an error state
-	Precision    string `json:"precision" yaml:"precision,omitempty"`
+	Type          string `json:"type" yaml:"type,omitempty"`                 // ValueDescriptor Type of property after transformations
+	ReadWrite     string `json:"readWrite" yaml:"readWrite,omitempty"`       // Read/Write Permissions set for this property
+	Minimum       string `json:"minimum" yaml:"minimum,omitempty"`           // Minimum value that can be get/set from this property
+	Maximum       string `json:"maximum" yaml:"maximum,omitempty"`           // Maximum value that can be get/set from this property
+	DefaultValue  string `json:"defaultValue" yaml:"defaultValue,omitempty"` // Default value set to this property if no argument is passed
+	Size          string `json:"size" yaml:"size,omitempty"`                 // Size of this property in its type  (i.e. bytes for numeric types, characters for string types)
+	Mask          string `json:"mask" yaml:"mask,omitempty"`                 // Mask to be applied prior to get/set of property
+	Shift         string `json:"shift" yaml:"shift,omitempty"`               // Shift to be applied after masking, prior to get/set of property
+	Scale         string `json:"scale" yaml:"scale,omitempty"`               // Multiplicative factor to be applied after shifting, prior to get/set of property
+	Offset        string `json:"offset" yaml:"offset,omitempty"`             // Additive factor to be applied after multiplying, prior to get/set of property
+	Base          string `json:"base" yaml:"base,omitempty"`                 // Base for property to be applied to, leave 0 for no power operation (i.e. base ^ property: 2 ^ 10)
+	Assertion     string `json:"assertion" yaml:"assertion,omitempty"`       // Required value of the property, set for checking error state.  Failing an assertion condition will mark the device with an error state
+	Precision     string `json:"precision" yaml:"precision,omitempty"`
+	FloatEncoding string `json:"floatEncoding,omitempty"` // FloatEncoding indicates the representation of floating value of reading.  It should be 'Base64' or 'eNotation'
```

Comment:
Omitempty in JSON is not used (we have MarshalJSON()). Also should be great to have the yaml tag with omitempty.